src/charm.py

```diff
@@ -174,12 +174,29 @@ def _process_wheel_file(self, event):  # noqa: C901
 
                 # Push wheel file to the container and extract it.
                 container.push(wheel_file, wheel_data, make_dirs=True)
-                container.exec(["apt-get", "update"]).wait()
-                container.exec(["apt-get", "install", "unzip"]).wait()
-                container.exec(["unzip", wheel_file, "-d", "/user_provided"]).wait()
+
+                # Rename wheel file to its original name and install it
+                container.exec(["mv", wheel_file, original_wheel_file]).wait()
+
+                _, error = container.exec(["pip", "install", original_wheel_file]).wait_output()
+                if error is not None and error.strip() != "" and not error.strip().startswith("WARNING"):
+                    logger.error(f"failed to install wheel file: {error}")
+                    raise ValueError("Invalid state: failed to install wheel file")
+
+                # Unpack wheel file
+                container.exec(["pip", "install", "wheel"]).wait()
```

Comment:
can you add this at the worker dependencies ?

Reply:
Updated, thanks!